### DISCUSSION TITLE
Question: Cloudflare proxy settings.

### DISCUSSION THREAD

**Discussion by Futodama** [plan-player-analytics/Plan#2411] [Archive]
I want to set up a Proxy via Cloudflare, how do I do this?
There are a lot of settings there and each one works in its own way.
If there is also a server on the IP that I use, then the task of setting up a proxy on the destination port becomes very difficult.
I do not know which setting to use, if using SRV record, then what arguments to apply?

**Comment by AuroraLS3** [comment]
That's not a question 🤔

**Comment by Futodama** [comment]
> That's not a question

I wrote it for a long time, and the page was updated.

**Comment by AuroraLS3** [comment]
I don't unfortunately use Cloudflare so I don't know how to set up their settings.

I know that:
- They use SRV records to set up the proxying rules for ports, eg specific address is routed to port on the IP in the A-record
- They have some sort of Proxy mode that hides real IP and routes traffic to your actual IP in the background

If you already have set their system up and it has https you can use this:
https://github.com/plan-player-analytics/Plan/wiki/SSL-Certificate-%28HTTPS%29-Set-Up#if-behind-a-proxy

**Comment by Futodama** [comment]
![image](https://user-images.githubusercontent.com/73364285/171806320-578c8cff-818b-426c-a711-31acbd4a994d.png)

In the settings of the "A" entry, it is not possible to insert the port to which the request should be sent. By default, it will only send to IP.

**Comment by Futodama** [comment]
It is possible to specify a specific port only for SRV

**Comment by AuroraLS3** [comment]
- A-record is for the IP
- SRV record is for the port when Proxied

**Comment by Futodama** [comment]
The problem is that if I set it up like this: plan.domaiun.ru and I specify the same thing in "Alternative_IP" my access does not work at all.

**Comment by Futodama** [comment]
That's not how it works for me. If you specify here the IP that the server has, then everything works. But again, it works without a Proxy.

    Alternative_IP: true
        Address: plan.domain.com

**Comment by AuroraLS3** [comment]
Sounds like something is wrong with the cloudflare configuration 🤔 

If you set Alternative_IP to plan.domain.com and `IP:<port>` redirects to `plan.domain.com` then everything on the Plan end is configured correctly

**Comment by Futodama** [comment]
"If you set Alternative_IP to plan.domain.com and IP:<port> redirects to plan.domain.com"

It works without problems, tested. If there is an IP address without specifying the domain, then everything works. But it does not work if installed on a proxy and "plan.domain.com ".

**Comment by AuroraLS3** [comment]
You might want to ask cloudflare support for assistance

**Comment by Futodama** [comment]
![image](https://user-images.githubusercontent.com/73364285/171809156-79d52ae3-db2c-49e3-9a71-7d046e43851a.png)

At the moment, the certificate from the plugin developer + IP in "Alternative_IP:"
And it works without fail.

**Comment by Futodama** [comment]
But I would like to configure it for the domain. It is unpleasant for me to see the IP.

**Comment by Futodama** [comment]
If I set the proxy mode, and set the IP, and then configure "plan.domain.com" will it work?

**Comment by AuroraLS3** [comment]
It sounds like cloudflare is not routing the traffic to the correct place - you already had everything on Plan side working when you commented:

> "If you set Alternative_IP to plan.domain.com and IP: redirects to plan.domain.com"
> 
> It works without problems, tested. If there is an IP address without specifying the domain, then everything works. But it does not work if installed on a proxy and "plan.domain.com ".

**Comment by Futodama** [comment]
> It sounds like cloudflare is not routing the traffic to the correct place - you already had everything on Plan side working when you commented:
> 
> > "If you set Alternative_IP to plan.domain.com and IP: redirects to plan.domain.com"
> > It works without problems, tested. If there is an IP address without specifying the domain, then everything works. But it does not work if installed on a proxy and "plan.domain.com ".

I can't configure via Proxy, but I can by certificate. I need a proxy to give access to administrators without showing IP.

**Comment by Futodama** [comment]
Otherwise, the whole point is plan.domain.com lost.

**Comment by AuroraLS3** [comment]
Please let me be clear: You need to ask Cloudflare for help with setting up plan.domain.com. https://support.cloudflare.com/hc/en-us/

I don't know cloudflare

**Comment by Futodama** [comment]
The direct purpose of installing letter access is not to show the IP to anyone. If you leave the IP, then the question arises of the vulnerability of the main host server.

**Comment by Futodama** [comment]
Once again, the server is on the hosting, and the domain is on the domain registrar. The registrar is not Cloudflare, and the server host is not Cloudflare. Cloudflare is here as a protection and proxy.

**Comment by AuroraLS3** [comment]
That is irrelevant if the cloudflare is the issue here - you need to ask their support for help with cloudflare issues

**Comment by Futodama** [comment]
> If I set the proxy mode, and set the IP, and then configure "plan.domain.com" will it work?

Will it work or not?

**Comment by AuroraLS3** [comment]
It will work if:

- You can access the Plan webserver through plan.domain.com
- It is https://

**Comment by AnttiMK** [comment]
Please see [this link.](https://developers.cloudflare.com/fundamentals/get-started/reference/network-ports/#network-ports-compatible-with-cloudflares-proxy) The Cloudflare proxy (by default) only proxies traffic that is destined to a specific port, i.e. if `example.com:8443` is requested, it will proxy the traffic to `backendIP:8443` via Cloudflare. The proxy configuration can't itself act as a reverse proxy, where it proxies traffic to a different port that is destined, i.e. it cannot proxy traffic destined to `example.com:80` (which is the default HTTP port) to `backendIP:12345`

The way you're supposed to do this is to have a reverse proxy machine/VPS/whatever running a web server on port 80 (and 443 for SSL), and then use the Cloudflare proxy to mask the IP address of that machine. Cloudflare's "normal orange cloud proxy" isn't really designed to be used as a reverse proxy. [Cloudflare Spectrum](https://www.cloudflare.com/products/cloudflare-spectrum/) is a reverse proxy product, but it's meant for enterprise use and definitely isn't free.
